main.go

```diff
@@ -88,108 +96,175 @@ func (c *DownloadPlugin) Run(cliConnection plugin.CliConnection, args []string)
+	pathVals := GetDirectoryContext(workingDir, paths, flagVals.File_flag)
+	for _, v := range pathVals {
+		v.RootWorkingDirectoryLocal = filepath.FromSlash(v.RootWorkingDirectoryLocal)
+	}
 
 	// ensure cf_trace is disabled, otherwise parsing breaks
 	if os.Getenv("CF_TRACE") == "true" {
 		fmt.Println("\nError: environment variable CF_TRACE is set to true. This prevents download from succeeding.")
 		return
 	}
 
-	// prevent overwriting files
-	if Exists(rootWorkingDirectoryLocal) && flagVals.OverWrite_flag == false {
-		fmt.Println("\nError: destination path", rootWorkingDirectoryLocal, "already Exists and is not an empty directory.\n\nDelete it or use 'cf download APP_NAME --overwrite'")
-		os.Exit(1)
-	}
+	// download files at each input path
+	for _, v := range pathVals {
+		// prevent overwriting files
+		if Exists(v.RootWorkingDirectoryLocal) && flagVals.OverWrite_flag == false {
+			fmt.Println("\nError: destination path", v.RootWorkingDirectoryLocal, "already exists.\n\nDelete it or rerun the command with the '--overwrite' flag.")
+			os.Exit(1)
+		}
 
-	cmdExec := cmd_exec.NewCmdExec()
-	parser = dir_parser.NewParser(cmdExec, appName, flagVals.Instance_flag, onWindows, flagVals.Verbose_flag)
-	dloader = downloader.NewDownloader(cmdExec, &wg, appName, flagVals.Instance_flag, rootWorkingDirectoryServer, flagVals.Verbose_flag, onWindows)
+		// remove files to be overwritten
+		if flagVals.OverWrite_flag {
+			err := os.RemoveAll(v.RootWorkingDirectoryLocal)
+			check(err, "Cannot remove "+v.RootWorkingDirectoryLocal+" for overwrite.")
+		}
 
-	// parse the directory
-	files, dirs := parser.ExecParseDir(startingPathServer)
+		parser = dir_parser.NewParser(cmdExec, appName, flagVals.Instance_flag, onWindows, flagVals.Verbose_flag)
+		dloader = downloader.NewDownloader(cmdExec, &wg, appName, flagVals.Instance_flag, v.RootWorkingDirectoryServer, flagVals.Verbose_flag, onWindows)
 
-	// stop consoleWriter
-	quit := make(chan int)
+		// stop consoleWriter
+		quit := make(chan int)
 
-	// disable consoleWriter if verbose
-	if flagVals.Verbose_flag == false {
-		go consoleWriter(quit)
-	}
+		// disable consoleWriter if verbose
+		if flagVals.Verbose_flag == false {
+			go consoleWriter(quit)
+		}
 
-	// Start the download
-	wg.Add(1)
-	dloader.Download(files, dirs, startingPathServer, rootWorkingDirectoryLocal, filterList)
+		if flagVals.File_flag {
+			// create directory for single file
+			err := os.MkdirAll(strings.TrimSuffix(v.RootWorkingDirectoryLocal, filepath.Base(v.RootWorkingDirectoryLocal)), 0755)
+			check(err, "Error D1: failed to create directory.")
+
+			// start download of single file
+			wg.Add(1)
+			dloader.DownloadFile(v.StartingPathServer, v.RootWorkingDirectoryLocal, &wg)
+		} else {
+			// parse the input directory
+			files, dirs := parser.ExecParseDir(v.StartingPathServer)
+
+			// start download of directory
+			wg.Add(1)
+			dloader.Download(files, dirs, v.StartingPathServer, v.RootWorkingDirectoryLocal, filterList)
```

Comment:
The `dloader` object stores a pointer to the wait group itself when initialized. It actually may not be necessary to pass it to `DownloadFile` either since the `dloader` already has it.